sdk/clientcore/annotation-processor-test/src/main/java/io/clientcore/annotation/processor/test/implementation/TestInterfaceClientService.java

```diff
@@ -78,6 +81,20 @@ Response<Foo> getFoo(@PathParam("key") String key, @QueryParam("label") String l
     Response<Void> deleteFoo(@PathParam("key") String key, @QueryParam("label") String label,
         @HeaderParam("Sync-Token") String syncToken);
 
+
+    @HttpRequestInformation(method = HttpMethod.GET, path = "foos", expectedStatusCodes = { 200 })
+    Response<FooListResult> listFooListResult(@HostParam("uri") String uri, RequestOptions requestOptions, Context context);
+
+    @HttpRequestInformation(method = HttpMethod.GET, path = "{nextLink}", expectedStatusCodes = { 200 })
+    Response<FooListResult> listNextFooListResult(@PathParam(value = "nextLink", encoded = true) String nextLink,
+                                                  RequestOptions requestOptions, Context context);
+
+    @HttpRequestInformation(method = HttpMethod.GET, path = "foos", expectedStatusCodes = { 200 })
+    Response<List<Foo>> listFoo(@HostParam("uri") String uri, RequestOptions requestOptions, Context context);
+
+    @HttpRequestInformation(method = HttpMethod.GET, path = "{nextLink}", expectedStatusCodes = { 200 })
+    Response<List<Foo>> listNextFoo(@PathParam(value = "nextLink", encoded = true) String nextLink,
+                                    RequestOptions requestOptions, Context context);
```

Comment:
I'm not sure if this is a nit or more important, but we shouldn't have `RequestOptions` and `Context` parameters on the same service interface methods - it should just be `RequestOptions`

Reply:
yes, nit. Let me remove Context.  I needed RequestOptions for passing in the ResponseBodyMode.